api/resolvers/item.js

```diff
@@ -1353,26 +1378,81 @@ export const createItem = async (parent, { forward, options, ...item }, { me, mo
       fee = sub.baseCost * ANON_FEE_MULTIPLIER + (item.boost || 0)
     }
   }
-  fee += imgFees;
+  fee += imgFees
 
-  ([item] = await serialize(
-    models.$queryRawUnsafe(
+  try {
+    ([item] = await serialize(
+      models.$queryRawUnsafe(
       `${SELECT} FROM create_item($1::JSONB, $2::JSONB, $3::JSONB, '${spamInterval}'::INTERVAL, $4::INTEGER[]) AS "Item"`,
       JSON.stringify(item), JSON.stringify(fwdUsers), JSON.stringify(options), uploadIds),
-    { models, lnd, me, hash, hmac, fee }
-  ))
+      { models, lnd, me, hash, hmac, fee }
+    ))
+  } catch (err) {
+    // post payment flow is only allowed for stackers, not anons since we can't show pending items to anons
+    if (!me || !(err instanceof InsufficientFundsError)) {
+      throw err
+    }
 
-  await createMentions(item, models)
+    // create invoice and insert as pending item
+    const invLimit = me ? INV_PENDING_LIMIT : ANON_INV_PENDING_LIMIT
+    const balanceLimit = USER_IDS_BALANCE_NO_LIMIT.includes(Number(me?.id)) ? 0 : me ? BALANCE_LIMIT_MSATS : ANON_BALANCE_LIMIT_MSATS
+    const description = 'Creating item on stacker.news'
+    // TODO: allow users to configure expiration
+    const expiresAt = datePivot(new Date(), { milliseconds: DEFAULT_INVOICE_TIMEOUT_MS })
+    // we don't do partial payments yet, users need to pay full amount via invoice
+    const mtokens = err.cost
+
+    const lndInv = await createInvoice({
+      description: me.hideInvoiceDesc ? undefined : description,
+      lnd,
+      mtokens,
+      expires_at: expiresAt
+    })
 
-  await enqueueDeletionJob(item, models)
+    let invoice
+    // need to use interactive tx here to set invoice.actionId = item.id
+    await models.$transaction(
+      async (tx) => {
+        // insert pending item
+        ([item] = await tx.$queryRawUnsafe(
+          `${SELECT} FROM create_item($1::JSONB, $2::JSONB, $3::JSONB, '${spamInterval}'::INTERVAL, $4::INTEGER[]) AS "Item"`,
+          JSON.stringify({ ...item, status: 'PENDING' }), JSON.stringify(fwdUsers), JSON.stringify(options), uploadIds))
+
+        // set required invoice data to update item on payment
+        const actionData = { cost: err.cost, credits: err.balance };
```

Comment:
If these fields need to be saved, might we do better storing them as columns with types and uniform naming?